runner.go

```diff
@@ -22,6 +24,9 @@ const (
 	ErrDead    = errors.ConstError("worker runner is not running")
 )
 
+// StartFunc is the type of the function that creates a worker.
+type StartFunc func(context.Context) (Worker, error)
```

Comment:
Does this need a new type? I'd be inclined to use an alias if it's just a cosmetic convenience.